include/cetl/any.hpp

```diff
@@ -0,0 +1,766 @@
+/// @file
+/// Defines the C++17 `std::any` type and several related entities.
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef CETL_ANY_HPP_INCLUDED
+#define CETL_ANY_HPP_INCLUDED
+
+#include "rtti.hpp"
+#include "pf17/cetlpf.hpp"
+#include "pf17/utility.hpp"
+#include "pf17/attribute.hpp"
+
+#include <algorithm>
+#include <initializer_list>
+
+#ifndef CETL_H_ERASE
+#    include "cetl/cetl.hpp"
+#endif
+
+namespace cetl
+{
+
+// Forward declarations
+template <std::size_t Footprint, bool Copyable, bool Movable, std::size_t Alignment>
+class any;
```

Comment:
This is misleading. Let's rename the header and type to `static_any` to avoid any confusion that this is a different implementation then the standard requires.

Reply:
`static` may not be a good choice if we extend this any with PMR support which we discussed in the past. Perhaps the following are better options:

- `any2`
- `configurable_any`
- `heapless_any` (works for PMR as well because we don't use the global heap directly)

Reply:
- `super_mega_any`
- `this_any_sponsored_by_progressive__progressive_will_save_you_hundreds_on_car_insurance`
- `less_evil_any`

## serious ones
`many` - sorta cheeky

`any_of` - what if we changed the template to take a variadic list of types and it does the `max(sizeof(...))` so the usage would look like `any_of<int, float, double> a{1};`?

Reply:
ok here's another inspired by your comment:

- `unbounded_variant`

>what if we changed the template to take a variadic list of types and it does the max(sizeof(...)) so the usage would look like any_of<int, float, double> a{1};?

If you need `variant`, you know where to find it. Perhaps you want an overload for `make_any`; we could add that later.

Reply:
unbounded_variant. love it

Reply:
`cetl::unbounded_variant` it is; no PF stuff anymore.